### PR TITLE
Move installation files into their own directory

### PR DESCRIPTION
**Pull request by lstein**
The root directory is getting cluttered. This PR moves all environment* and requirements* files into a subdirectory named `install`.

- environment*.yml and requirements* files all moved into a new subdirectory named install
- updated documentation accordingly

I ran through the entire Linux installation instructions on a clean system, and everything seems to work OK.

### CODE REVIEW

File: docs/installation/INSTALL_LINUX.md [hunk 1]
```diff
@@ -59,17 +59,15 @@ This will create InvokeAI folder where you will follow the rest of the steps.
 (invokeai) ~/InvokeAI$ python3 scripts/preload_models.py
 ```
 
-    !!! note
```
Comment (tildebyte) [inline docs/installation/INSTALL_LINUX.md:62]: @lstein This syntax is used by the job which transforms the MD into the HTML for Pages

Comment (lstein) [reply docs/installation/INSTALL_LINUX.md:62]: Sure, will put back.

How does this reorganization of the requirements files interact with your one-click installer script. And how is that going?

Comment (tildebyte) [reply docs/installation/INSTALL_LINUX.md:62]: > interact with your one-click installer script

All of the requirements and environment files will be deleted

> how is that going?

"Works For Me" - but I think we all know how dangerous that is :rofl: 

TL;DR - When I feel confident enough, I'm going to open a draft PR. At that point, I will need testing on as large a platform (platform == OS/CPU/(GPU|!GPU) matrix as humanly possible, along with as much of the preflight checks as people have patience for.

Also of note: I don't think that this will be the final stop on this particular subway - that would be a stand-alone Python application per-platform - but Python application bundling is dark magic, and AFAICT there's no ***standard*** (Conda bundles or whatever they're called *are not* a "standard")

**Comment by colemickens** [comment]
I don't get it, this moves away from like the one single sane relatively widely followed python conventions?

**Comment by tildebyte** [comment]
There really isn't one sane Python convention (which frankly is the problem) other than setup.py, which is at least half-deprecated

**Comment by benjifriedman** [comment]
> There really isn't one sane Python convention (which frankly is the problem) other than setup.py, which is at least half-deprecated

wait what!? why? how did that happen?

**Comment by tildebyte** [comment]
OTOH, I saw this (again) today, and remembered why I thought "deprecated" - this is from a plain old `pip install` with latest 'pip' (emphasis added):

```
DEPRECATION: pathtools is being installed using the ***legacy 'setup.py install' method***, because it does not
have a 'pyproject.toml' and the 'wheel' package is not installed. pip 23.1 will enforce this behaviour change.
```

FWIW current 'pip' is 22.x

P.S. weirdly, 'wheel' IS installed...
